docs/articles/documentation/test-api/obtaining-data-from-the-client.md

```diff
@@ -0,0 +1,198 @@
+from the original function and new `options` that overwrite the original ones.
+
+The sample below shows how to overwrite the client function options to use a different selector.
+
+```js
+import { Selector, ClientFunction } from 'testcafe';
+
+const getThirdElemByClass = Selector(cl => document.getElementsByClassName(cl), { index: 2 });
+const getThirdElemWidth   = ClientFunction(cl => getThirdElemByClass(cl).width, {
+     dependencies: { getThirdElemByClass }
+});
+
+const getFourthElemByClass  = getThirdElemByClass.with({ index: 3 });
+const getFourthElemWidth    = getThirdElemWidth.with({
+    dependencies: { getFourthElemByClass }
+});
+```
+
+## One-Time Client Code Execution
+
+To create a client function and immediately execute it without saving,
+use the `eval` method of the [test controller](test-code-structure.md#test-controller).
+
+```text
+t.eval( fn [, options] )
+```
+
+Parameter              | Type     | Description
+---------------------- | -------- | --------------------------------------------------------------------------
+`fn`                   | Function | A function to be executed on the client side.
+`options` *(optional)* | Object   | See [Options](#options).
+
+The following example shows how to get the document's URI with `t.eval`.
+
+```js
+test('My Test', async t => {
+    const docURI = await t.eval(() => document.documentURI);
+});
+```
+
+## Calling Client Functions from Node.js Callbacks
+
+Client functions need access to the [test controller](test-code-structure.md#test-controller) to be executed.
+When called right from the test function, they implicitly obtain the test controller.
+
+However, if you need to call a client function from a Node.js callback that fires during the test run,
+you will have to manually bind this function to the test controller.
+
+Use the [boundTestRun](#optionsboundtestrun) option for this.
+
+```js
+import { fs } from 'fs';
+import { expect } from 'chai';
+import { ClientFunction } from 'testcafe';
+
+fixture `My fixture`
+    .page('http://www.example.com/');
+
+const getDataFromClient = ClientFunction(() => getSomeData());
+
+test('Check client data', async t => {
+    const boundFunc = getDataFromClient.with({ boundTestRun: t });
+
+    const equal = await new Promise(resolve => {
+        fs.readFile('/home/user/tests/reference/clientData.json', (err, data) => {
+            resolve(JSON.stringify(await boundFunc()) === data);
+        });
+    });
+
+    expect(equal).to.be.true;
+});
+```
+
+This approach only works for Node.js callbacks that fire during the test run. To ensure that the test function
+does not finish before the callback is executed, suspend the test until the callback fires. You can do this
+by introducing a promise and synchronously waiting for it to complete as shown in the example above.
+
+## Limitations
+
+* You cannot use generators or `async/await` syntax within client functions.
```

Comment:
1) You cannot make separate consequent blockquotes (which we render as notes and warnings) with the redcarpet processor. It merges them into one.
2) This fact is not so bad because the header seem to be enough to emphasize this piece of information. A lot of additional 'ink' (warning borders and icons) will hardly be of great help.